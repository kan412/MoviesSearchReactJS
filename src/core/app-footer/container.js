import React from 'react';

class AppFooter extends React.PureComponent {
  render() {
    return (
      <footer>
        <p>&copy; All rights Reserved</p>
      </footer>
    );
  }
}


export default AppFooter;
