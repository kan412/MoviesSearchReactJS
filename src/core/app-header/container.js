import React from 'react';

class AppHeader extends React.PureComponent {
  render() {
    return (
      <header>
        <div className="header-inner">
          <a href="/">
            <h1 className="logo">
              Netflix
              <span>roulette</span>
            </h1>
          </a>
        </div>
      </header>
    );
  }
}


export default AppHeader;
