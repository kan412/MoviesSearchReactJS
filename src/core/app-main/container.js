import React from 'react';
import MoviesSearch from '../../feature/movies-search';
import { moviesList } from '../../shared';

class AppMain extends React.PureComponent {
  render() {
    // const mList = JSON.stringify(moviesList);
    return (
      <MoviesSearch results={moviesList} />
    );
  }
}


export default AppMain;
