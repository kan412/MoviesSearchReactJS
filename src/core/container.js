import React from 'react';
import AppHeader from './app-header';
import AppMain from './app-main';
import AppFooter from './app-footer';
import ErrorBoundary from '../shared/error-boundary';
import './component.css';

class App extends React.PureComponent {
  render() {
    return (
      <ErrorBoundary>
        <AppHeader />
        <AppMain />
        <AppFooter />
      </ErrorBoundary>
    );
  }
}

export default App;
