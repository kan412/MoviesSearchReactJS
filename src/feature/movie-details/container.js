import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import MovieCard from '../../shared/movie-card';
import getYear from '../../shared/utils';
import config from '../../../config';
import styles from './component.css';

class MovieDetailsContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      movie: {},
      relatedMovies: [],
    };
  }

  componentDidMount() {
    const { match } = this.props;
    const queryString = `${config.API_URL}/${match.params.id}`;

    fetch(queryString)
      .then(res => res.json())
      .then(mov => this.setState({ movie: mov }));

    const { movie } = this.state;
    const relatedMoviesQueryString = `${config.API_URL}?searchBy=genres&filter=${movie.genres.join(',')}`;

    fetch(relatedMoviesQueryString)
      .then(res => res.json())
      .then(movies => this.setState({ relatedMovies: movies.data }));
  }

  handleMovieClick = (movie) => {
    const url = `/film/`;
    return <Redirect to={url} />;
  };

  render() {
    console.log(typeof this.handleMovieClick);
    const { movie, relatedMovies } = this.state;
    const mlist = relatedMovies.map(mov => <MovieCard key={mov.id} data={mov} onMovieClick={this.handleMovieClick} />);

    return (
      <React.Fragment>
        <div className={styles['movie-summary']}>
          <div className={styles.image}>
            <img src={movie.poster_path} width="250px" alt={movie.title} />
          </div>

          <div className={styles.details}>
            <h1 className={styles.title}>
              {movie.title}
              <span className={styles.rating}>{movie.vote_average}</span>
            </h1>
            <div className={styles.meta}>
              <span className={styles['release-year']}>
                <span>{getYear(movie.release_date)}</span>
                Year
              </span>
              <span className={styles.duration}>
                <span>{movie.runtime}</span>
                 Mins
              </span>
            </div>

            <div className={styles.description}>
              {movie.overview}
            </div>
          </div>
        </div>

        <div className={styles['search-results-header']}>
          <div className={styles['search-results-header-inner']}>
            <span>Films by  </span>
            { movie.genres }
            <span>  genre </span>
          </div>
        </div>

        <div className={styles['search-results-container']}>
          { (mlist.length > 0) ? mlist : 'No Items Found' }
        </div>
      </React.Fragment>
    );
  }
}


MovieDetailsContainer.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }).isRequired,
  }).isRequired,
};

export default MovieDetailsContainer;
