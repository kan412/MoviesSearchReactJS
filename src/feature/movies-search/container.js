import React from 'react';
import MovieCard from '../../shared/movie-card';

class MoviesSearch extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      data: this.props.results,
      searchby: 'title',
      sortby: 'releasedate',
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    const inputValue = document.getElementById('search-movies').value;

    this.setState(state => ({
      data: state.data.filter(movie => movie.title === inputValue),
    }));
  }

  render() {
    const { data } = this.state;
    const movies = data.map(movie => <MovieCard key={movie.id} data={movie} />);

    return (
      <React.Fragment>
        <form className="search-form">
          <label htmlFor="search-movies">
            <span>FIND YOUR MOVIE</span>
            <input type="text" id="search-movies" />
            <button type="button" onClick={this.handleClick}>Search</button>
          </label>
          <div className="search-filter">
            Search by
            <span className="buttons">
              <button type="button">Title</button>
              <button type="button">Genre</button>
            </span>
          </div>
        </form>

        <div className="search-results-header">
          <div className="search-results-header-inner">
            <div className="search-results-found">
              { movies.length }
              <span>movies found</span>
            </div>
            <div className="sort-filter">
              Sort by
              <span className="buttons">
                <button type="button">Release Date</button>
                <button type="button">Rating</button>
              </span>
            </div>
          </div>
        </div>

        <div className="search-results-container">
          { movies }
        </div>
      </React.Fragment>
    );
  }
}

// MoviesSearch.propTypes = {
//   results: PropTypes.
// };

export default MoviesSearch;
