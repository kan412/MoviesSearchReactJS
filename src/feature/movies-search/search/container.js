import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './component.css';
import { SEARCH_BY } from '../../../shared';
import { searchBY, updSearchTerm } from '../../../core/store/actions';


export class SearchContainer extends React.Component {
  static propTypes = {
    filterToggle: PropTypes.func.isRequired,
    updateSearchTerm: PropTypes.func.isRequired,
  }

  constructor(props) {
    super(props);

    this.searchInput = React.createRef();

    this.state = {
      searchBy: SEARCH_BY.TITLE,
    };
  }

  handleClick = () => {
    const inputValue = this.searchInput.current.value;
    const { filterToggle, updateSearchTerm } = this.props;
    const { searchBy } = this.state;
    filterToggle(searchBy);
    updateSearchTerm(inputValue);
  };

  handleFilter = (e) => {
    this.setState({ searchBy: e.target.value });
  };

  render() {
    const { searchBy } = this.state;
    return (
      <div className={styles['search-form']}>
        <label htmlFor="search-movies">
          <span>FIND YOUR MOVIE</span>
          <input type="text" ref={this.searchInput} id="search-movies" />
          <button type="button" onClick={this.handleClick} id={styles['search-button']}>Search</button>
        </label>
        <div className={styles['search-filter']}>
          Search by
          <span className={styles['split-buttons']}>
            <button type="button" onClick={this.handleFilter} value={SEARCH_BY.TITLE} className={(searchBy === SEARCH_BY.TITLE) ? styles.active : ''}>Title</button>
            <button type="button" onClick={this.handleFilter} value={SEARCH_BY.GENRE} className={(searchBy === SEARCH_BY.GENRE) ? styles.active : ''}>Genre</button>
          </span>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  updateSearchTerm: (value) => {
    dispatch(updSearchTerm(value));
  },
  filterToggle: (value) => {
    dispatch(searchBY(value));
  },
});

export default connect(null, mapDispatchToProps)(SearchContainer);
