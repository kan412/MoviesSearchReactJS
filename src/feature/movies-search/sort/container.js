import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import styles from './component.css';
import { SORT_BY } from '../../../shared';
import { sortBY } from '../../../core/store/actions';

export class SortContainer extends React.Component {
  static propTypes = {
    sortToggle: PropTypes.func.isRequired,
    sortBy: PropTypes.string.isRequired,
  }

  handleSort = (e) => {
    const { sortToggle } = this.props;
    sortToggle(e.target.value);
  }

  render() {
    const { sortBy } = this.props;
    return (
      <div className={styles['search-results-sort']}>
        Sort by
        <span className={styles['split-buttons']}>
          <button type="button" onClick={this.handleSort} value={SORT_BY.YEAR} className={(sortBy === SORT_BY.YEAR) ? styles.active : ''}>Release Date</button>
          <button type="button" onClick={this.handleSort} value={SORT_BY.RATING} className={(sortBy === SORT_BY.RATING) ? styles.active : ''}>Rating</button>
        </span>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  sortBy: state.sortBy,
});


export const mapDispatchToProps = dispatch => ({
  sortToggle: (value) => {
    dispatch(sortBY(value));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(SortContainer);
