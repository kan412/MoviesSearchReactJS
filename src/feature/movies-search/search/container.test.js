import React from 'react';
import { shallow } from 'enzyme';
import { SEARCH_BY } from '../../../shared';
import { SearchContainer, mapDispatchToProps } from './container';

describe('Search', () => {
  it('renders properly', () => {
    const component = shallow(<SearchContainer />);
    expect(component).toMatchSnapshot();
  });

  it('default state of searchBy is title', () => {
    const component = shallow(<SearchContainer />);
    expect(component.state().searchBy).toBe(SEARCH_BY.TITLE);
  });

  it('handleClick', () => {
    const handleFilterToggle = jest.fn();
    const handleUpdateSearchTerm = jest.fn();
    const component = shallow(
      <SearchContainer
        filterToggle={handleFilterToggle}
        updateSearchTerm={handleUpdateSearchTerm}
      />,
    );
    const button = component.find('button').at(0);
    button.simulate('click');
    expect(handleFilterToggle).toHaveBeenCalled();
    expect(handleUpdateSearchTerm).toHaveBeenCalled();
  });

  it('mapDispatchToProps', () => {
    const dispatch = jest.fn();
    const inputValue = 'Matrix';
    const filterValue = SEARCH_BY.TITLE;
    mapDispatchToProps(dispatch).updateSearchTerm(inputValue);
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'UPDATE_SEARCH_TERM', payload: 'Matrix' });

    mapDispatchToProps(dispatch).filterToggle(filterValue);
    expect(dispatch.mock.calls[1][0]).toEqual({ type: 'SEARCH_BY', payload: 'title' });
  });

  it('update searchBy state on filter button click', () => {
    const component = shallow(<SearchContainer />);
    const button = component.find('.split-buttons').children().first();
    button.simulate('click', {
      target: {
        value: SEARCH_BY.TITLE,
      },
    });
    expect(component.state().searchBy).toBe(SEARCH_BY.TITLE);
  });
});
