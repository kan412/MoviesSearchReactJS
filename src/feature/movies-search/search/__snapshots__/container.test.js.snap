// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders properly 1`] = `
<div
  className="search-form"
>
  <label
    htmlFor="search-movies"
  >
    <span>
      FIND YOUR MOVIE
    </span>
    <input
      id="search-movies"
      type="text"
    />
    <button
      id="search-button"
      onClick={[Function]}
      type="button"
    >
      Search
    </button>
  </label>
  <div
    className="search-filter"
  >
    Search by
    <span
      className="split-buttons"
    >
      <button
        className="active"
        onClick={[Function]}
        type="button"
        value="title"
      >
        Title
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="genres"
      >
        Genre
      </button>
    </span>
  </div>
</div>
`;
